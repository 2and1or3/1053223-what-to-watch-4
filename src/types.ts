interface FilmType {
  id: number;
  title: string;
  preview: string;
  background: string;
  backgroundColor: string;
  cover: string;
  genre: string;
  isFavorite: boolean;
  src: string;
  release: string;
  description: string;
  poster: string;
  duration: number;
  rating: number;
  voiceCount: number;
  director: string;
  actors: string[];
}

interface CommentType {
  id: string;
  author: string;
  date: string;
  description: string;
  rate: string;
}

interface GenreType {
  id: string;
  title: string;
}

type RefType = () => void | {current: React.ReactNode};

export {FilmType, CommentType, RefType, GenreType};
