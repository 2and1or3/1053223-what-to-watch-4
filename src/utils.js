import {AppRoute, MONTHS} from './consts';

const createVideoMock = (element) => {
  if (element.type === `video`) {
    return element;
  }

  return null;
};


const extend = (a, b) => {
  return Object.assign({}, a, b);
};

const adapterToLocalFilms = (films) => {

  const localFilms =
    films
    .slice()
    .map((film) => ({
      id: film.id.toString(),
      isFavorite: film.is_favorite,
      title: film.name,
      poster: film.preview_image,
      preview: film.preview_video_link,
      src: film.video_link,
      background: film.background_image,
      backgroundColor: film.background_color,
      cover: film.poster_image,
      genre: film.genre.toLowerCase(),
      release: film.released.toString(),
      description: film.description,
      rating: film.rating,
      voiceCount: film.scores_count,
      duration: film.run_time,
      director: film.director,
      actors: film.starring.slice(),
    }));

  return localFilms;
};

const adapterToLocalComments = (comments) => {

  const localComments =
  comments
  .map((comment) => ({
    id: comment.id.toString(),
    author: comment.user.name,
    date: comment.date,
    description: comment.comment,
    rate: comment.rating.toString(),
  }));

  return localComments;
};

const GetPath = {
  filmReview: (id) => `${AppRoute.FILM}${id}${AppRoute.REVIEW}`,
  filmPlayer: (id) => `${AppRoute.FILM}${id}${AppRoute.PLAYER}`,
  filmDetails: (id) => `${AppRoute.FILM}${id}`,
};

const getDateFormat = (date) => {
  const data = new Date(date);
  const month = data.getMonth();
  const day = data.getDate();
  const year = data.getFullYear();

  return `${MONTHS[month]} ${day} ${year}`;
};

const getWatchTimeFormat = (seconds) => {

  let hours = Math.floor(seconds / 3600);
  let min = Math.floor((seconds - hours * 3600) / 60);
  let sec = seconds - hours * 3600 - min * 60;

  hours = hours.toString();
  min = min.toString().padStart(2, `0`);
  sec = sec.toString().padStart(2, `0`);

  return `${hours}:${min}:${sec}`;
};

const capitalize = (str) => {
  str = str.toLowerCase();
  let words = str.split(` `);

  words = words.map((word) => word[0].toUpperCase() + word.slice(1));
  str = words.join(` `);

  return str;
};

const getUniqueGenres = (genres) => {
  let uniqueGenres = [];

  genres.forEach((genre) => {
    const isRepeated = uniqueGenres.some((uniqueGenre) => uniqueGenre === genre);

    if (!isRepeated) {
      uniqueGenres.push(genre);
    }
  });


  const localUniqueGenres = uniqueGenres.map((uniqueGenre) => ({
    id: uniqueGenre,
    title: capitalize(uniqueGenre),
  }));

  return localUniqueGenres;
};

const noop = () => {};

export {createVideoMock, adapterToLocalFilms, extend, adapterToLocalComments, GetPath, getDateFormat, getWatchTimeFormat, getUniqueGenres, noop};
