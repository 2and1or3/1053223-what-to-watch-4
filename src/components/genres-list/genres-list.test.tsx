import * as React from "react";
import * as renderer from "react-test-renderer";

import GenresList from './genres-list';
import {GenreType} from '../../types';
import {noop} from '../../utils';

const genres: GenreType[] = [
  {
    id: `id1`,
    title: `id1`,
  },
  {
    id: `id2`,
    title: `id2`,
  },
];

it(`Render GenresList component`, () => {
  const tree = renderer
    .create(<GenresList activeItem = {`all`} onLinkClick = {noop} onTargetClick = {noop} genres = {genres}/>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

// npm run test.jest -- components/genres-list/genres-list.test
