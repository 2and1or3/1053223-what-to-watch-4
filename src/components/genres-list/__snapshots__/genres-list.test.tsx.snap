// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render GenresList component 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item "
    id="id1"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      id1
    </a>
  </li>
  <li
    className="catalog__genres-item "
    id="id2"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      id2
    </a>
  </li>
</ul>
`;
